sale_order_type/models/sale.py

```diff
@@ -73,6 +74,18 @@ def _prepare_invoice(self):
             res["sale_type_id"] = self.type_id.id
         return res
 
+    @api.constrains("company_id")
+    def _check_so_type_company(self):
+        for record in self.filtered("type_id.company_id"):
```

Comment:
```suggestion
        if self.filtered(lambda r: r.type_id.company_id != r.company_id):
```
Maybe something like this

Reply:
My point also that, if user do not select type or type has no company specified, we don't want to check.
On your point, it still check, isn't it?

Reply:
You are right, Maybe in that case we can check something like this


```suggestion
        if self.filtered(lambda r: r.type_id.company_id and r.company_id and r.type_id.company_id != r.company_id):
```

Reply:
Done, thank you!